src/Microsoft.OpenApi/Validations/Rules/RuleHelpers.cs

```diff
@@ -0,0 +1,38 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. 
+
+using System;
+
+namespace Microsoft.OpenApi.Validations.Rules
+{
+    internal static class RuleHelpers
+    {
+        /// <summary>
+        /// Input string must be in the format of an email address
+        /// </summary>
+        /// <param name="input">The input string.</param>
+        /// <returns></returns>
+        public static bool IsEmailAddress(this string input)
```

Comment:
>IsEmailAddress [](start = 27, length = 14)

The accepted answer here looks reasonable to me:

[https://stackoverflow.com/questions/1365407/c-sharp-code-to-validate-email-address](https://stackoverflow.com/questions/1365407/c-sharp-code-to-validate-email-address)

It is though a little more forgiving than it has to be, but I believe it is strict enough for our purpose.